### ISSUE TITLE
pilz_cartesian_limits.yaml overwrites joint_limits.yaml when using moveit_configs_builder.py 

### ISSUE THREAD

**Issue by ipa-cmh** [ros-planning/moveit2#1691]
### Description

When using an MSA generated config package, the parameters stored in pilz_cartesian_limits.yaml overwrite the parameters stored in joint_limits.yaml.

### Your environment
* ROS Distro: Humble
* OS Version: Ubuntu 22.04
* Binary install

### Steps to reproduce
Create a msa package and add the appropriate configuration YAMLs.
```
config/
- joint_limits.yaml
- pilz_cartesian_limits.yaml
```

With the follwoing content:
```
# Joints limits
#
# Sources:
#
#  - Universal Robots e-Series, User Manual, UR5e, Version 5.8
#    https://s3-eu-west-1.amazonaws.com/ur-support-site/69091/99404_UR5e_User_Manual_en_Global.pdf
#  - Support > Articles > UR articles > Max. joint torques
#    https://www.universal-robots.com/articles/ur-articles/max-joint-torques
#    retrieved: 2020-06-16, last modified: 2020-06-09
joint_limits:
  shoulder_pan_joint:
    # acceleration limits are not publicly available
    has_acceleration_limits: true
    has_effort_limits: true
    has_position_limits: true
    has_velocity_limits: true
    max_effort: 150.0
    max_position: 3.14159265359
    max_velocity: 1.57079632679
    min_position: -3.14159265359
    max_acceleration: 3.49
  shoulder_lift_joint:
    # acceleration limits are not publicly available
    has_acceleration_limits: true
    has_effort_limits: true
    has_position_limits: true
    has_velocity_limits: true
    max_effort: 150.0
    max_position: 3.14159265359
    max_velocity: 1.57079632679
    min_position: -3.14159265359
    max_acceleration: 3.49
  elbow_joint:
    # acceleration limits are not publicly available
    has_acceleration_limits: true
    has_effort_limits: true
    has_position_limits: true
    has_velocity_limits: true
    max_effort: 150.0
    # we artificially limit this joint to half its actual joint position limit
    # to avoid (MoveIt/OMPL) planning problems, as due to the physical
    # construction of the robot, it's impossible to rotate the 'elbow_joint'
    # over more than approx +- 1 pi (the shoulder lift joint gets in the way).
    #
    # This leads to planning problems as the search space will be divided into
    # two sections, with no connections from one to the other.
    #
    # Refer to https://github.com/ros-industrial/universal_robot/issues/265 for
    # more information.
    max_position: 1.57079632679
    max_velocity: 1.57079632679
    min_position: -1.57079632679
    max_acceleration: 3.49
  wrist_1_joint:
    # acceleration limits are not publicly available
    has_acceleration_limits: true
    has_effort_limits: true
    has_position_limits: true
    has_velocity_limits: true
    max_effort: 28.0
    max_position: 3.14159265359
    max_velocity: 1.57079632679
    min_position: -3.14159265359
    max_acceleration: 3.49
  wrist_2_joint:
    # acceleration limits are not publicly available
    has_acceleration_limits: true
    has_effort_limits: true
    has_position_limits: true
    has_velocity_limits: true
    max_effort: 28.0
    max_position: 3.14159265359
    max_velocity: 1.57079632679
    min_position: -3.14159265359
    max_acceleration: 3.49
  wrist_3_joint:
    # acceleration limits are not publicly available
    has_effort_limits: true
    has_position_limits: true
    has_velocity_limits: true
    max_effort: 28.0
    max_position: 3.14159265359
    max_velocity: 1.57079632679
    min_position: -3.14159265359
    has_acceleration_limits: true
    max_acceleration: 3.49
```

Cartesian limits
```
cartesian_limits:
  max_trans_vel: 1.0
  max_trans_acc: 2.25
  max_trans_dec: -5.0
  max_rot_vel: 1.57
```

### Expected behaviour
The parameters in both files should be joined into robot_description_planning when using to_dict call, should result in:
```
 'robot_description_planning': {'cartesian_limits': {'max_rot_vel': 1.57,
                                                     'max_trans_acc': 2.25,
                                                     'max_trans_dec': -5.0,
                                                     'max_trans_vel': 1.0},
                                'joint_limits': {'elbow_joint': {'has_acceleration_limits': True,
                                                                 'has_effort_limits': True,
                                                                 'has_position_limits': True,
                                                                 'has_velocity_limits': True,
                                                                 'max_acceleration': 3.49,
                                                                 'max_effort': 150.0,
                                                                 'max_position': 1.57079632679,
                                                                 'max_velocity': 1.57079632679,
                                                                 'min_position': -1.57079632679},
                                                 'shoulder_lift_joint': {'has_acceleration_limits': True,
                                                                         'has_effort_limits': True,
                                                                         'has_position_limits': True,
                                                                         'has_velocity_limits': True,
                                                                         'max_acceleration': 3.49,
                                                                         'max_effort': 150.0,
                                                                         'max_position': 3.14159265359,
                                                                         'max_velocity': 1.57079632679,
                                                                         'min_position': -3.14159265359},
                                                 'shoulder_pan_joint': {'has_acceleration_limits': True,
                                                                        'has_effort_limits': True,
                                                                        'has_position_limits': True,
                                                                        'has_velocity_limits': True,
                                                                        'max_acceleration': 3.49,
                                                                        'max_effort': 150.0,
                                                                        'max_position': 3.14159265359,
                                                                        'max_velocity': 1.57079632679,
                                                                        'min_position': -3.14159265359},
                                                 'wrist_1_joint': {'has_acceleration_limits': True,
                                                                   'has_effort_limits': True,
                                                                   'has_position_limits': True,
                                                                   'has_velocity_limits': True,
                                                                   'max_acceleration': 3.49,
                                                                   'max_effort': 28.0,
                                                                   'max_position': 3.14159265359,
                                                                   'max_velocity': 1.57079632679,
                                                                   'min_position': -3.14159265359},
                                                 'wrist_2_joint': {'has_acceleration_limits': True,
                                                                   'has_effort_limits': True,
                                                                   'has_position_limits': True,
                                                                   'has_velocity_limits': True,
                                                                   'max_acceleration': 3.49,
                                                                   'max_effort': 28.0,
                                                                   'max_position': 3.14159265359,
                                                                   'max_velocity': 1.57079632679,
                                                                   'min_position': -3.14159265359},
                                                 'wrist_3_joint': {'has_acceleration_limits': True,
                                                                   'has_effort_limits': True,
                                                                   'has_position_limits': True,
                                                                   'has_velocity_limits': True,
                                                                   'max_acceleration': 3.49,
                                                                   'max_effort': 28.0,
                                                                   'max_position': 3.14159265359,
                                                                   'max_velocity': 1.57079632679,
                                                                   'min_position': -3.14159265359}}},

```

### Actual behaviour
The parameters in pilz_cartesian_limits.yaml overwrite joint_limits.yaml, resulting in:
```
 'robot_description_planning': {'cartesian_limits': {'max_rot_vel': 1.57,
                                                     'max_trans_acc': 2.25,
                                                     'max_trans_dec': -5.0,
                                                     'max_trans_vel': 1.0}},
```

### Backtrace or Console output

Use [gist.github.com](gist.github.com) to copy-paste the console output or segfault backtrace using gdb.

### Workaround
Put joint limits in pilz_cartesian_limits.yaml ....

**Comment by JafarAbdi** [comment]
Hi @ipa-cmh,

I can't reproduce it locally, do you mind uploading the generated moveit config package and share the link?

Thanks!

**Comment by ipa-cmh** [comment]
Hi @JafarAbdi,

I did some digging and it seems to be the `MoveItConfigs.to_dict()` function.
After reviewing the current source in master, this seems to have been solved with #1571 and #1583.
This is just not yet in the humble released package.

If necessary, you can use this repo https://github.com/ipa-rar/ur5e_cell/tree/ur5e-working-with-msa for reproduction.
